sklearn/model_selection/_split.py

```diff
@@ -637,6 +637,121 @@ def split(self, X, y, labels=None):
         """
         return super(StratifiedKFold, self).split(X, y, labels)
 
+class HomogeneousTimeSeriesCV(_BaseKFold):
```

Comment:
It is convenient to make `HomogeneousTimeSeriesCV ` subclass `_BaseKFold` since `__init__ ` in `_BaseKFold` can help `HomogeneousTimeSeriesCV` to check whether input parameter `n_folds` is valid.

However, in order to make `HomogeneousTimeSeriesCV` work, `HomogeneousTimeSeriesCV` need to override function `split` which is defined in its superclass `_BaseKFold` and its super-superclass `BaseCrossValidator` due to their implementation detail of `split`.

I don't think override `split` in `HomogeneousTimeSeriesCV ` is a good solution, can @rvraghav93 provide some suggestions on this?